NanoProd/config/skim_hzgamma.yaml

```diff
@@ -0,0 +1,109 @@
+  other_trees:
+    - LuminosityBlocks
+    - Runs
+  sel_ref: selection
+  column_filters:
+    - drop Jet_btagDeepB
+    - drop Jet_btagDeepCvB
+    - drop Jet_btagDeepCvL
+    - drop Jet_btagCSVV2
+    - drop Jet_qgl
+    - drop Jet_neHEF
+    - drop Jet_chFPV0EF
+    - drop Jet_hfsigmaEtaEta
+    - drop Jet_hfsigmaPhiPhi
+    - drop Jet_hfcentralEtaStripSize
+    - drop Jet_hfadjacentEtaStripsSize
+    - drop Jet_nElectrons
+    - drop Jet_electronIdx1
+    - drop Jet_electronIdx2
+    - drop Electron_sieie
+    - drop Electron_jetRelIso
+    - drop Electron_r9
+    - drop Electron_jetPtRelv2
+    - drop Electron_hoe
+    - drop Electron_dr03EcalRecHitSumEt
+    - drop Electron_dr03HcalDepth1TowerSumEt
+    - drop Electron_jetNDauCharged
+    - drop Electron_dr03TkSumPt
+    - drop Electron_dr03TkSumPtHEEP
+    - drop Electron_tightCharge
+    - drop Electron_lostHits
+    - drop Electron_cutBased_HEEP
+    - drop Muon_dxybs
+    - drop Muon_segmentComp
+    - drop Muon_jetRelIso
+    - drop Muon_jetPtRelv2
+    - drop Muon_nStations
+    - drop Muon_jetNDauCharged
+    - drop ^(n|)Tau(_.*|)$
+    - drop ^(n|)boostedTau(_.*|)$
+    # - drop ^(n|)SV(_.*|)$
+    - drop ^(n|)OtherPV(_.*|)$
+    - drop Photon_genPartIdx
+    - drop Photon_haloTaggerMVAVal
+    - drop Photon_hasConversionTracks
+    - drop Photon_seedGain
+    - drop Photon_seediEtaOriX
+    - drop Photon_seediPhiOriY
+    - drop Photon_x_calo
+    - drop Photon_y_calo
+    - drop Photon_z_calo
+    - drop ^(n|)LowPtElectron(_.*|)$
+    - drop ^(n|)SoftActivityJet(.*|)$
+    - drop ^(n|)IsoTrack(_.*|)$
+    - drop ^(n|)GenDressedLepton(_.*|)$
+    - drop ^(n|)GenIsolatedPhoton(_.*|)$
+    - drop ^(n|)GenProton(_.*|)$
+    - drop ^(n|)Proton(_.*|)$
+    - drop ^(n|)PPSLocalTrack(_.*|)$
+    - drop ^L1_.*$
+    - drop ^HLT_.*$
+    # - keep ^HLT_.*Tau.*$
+    # - keep ^HLT_PFMET.*$
+    # - keep ^HLT_PFHT.*$
+    - keep ^HLT_(|Iso|Old)(|Tk)Mu[1-9][0-9]+.*$
+    - keep ^HLT_Ele[1-9][0-9]+.*$
+    - drop ^HLT_Ele.*CaloId.*$
+    # - keep ^HLT_Ele[1-9][0-9]+.*CaloId.*PFJet.*$
+    # - keep ^HLT_Photon[0-9]+_TwoProngs[0-9]+$
+    # - keep ^HLT_Quad(|PF)Jet.*$
+    # - keep ^HLT_MonoCentralPFJet.*$
+    # - keep ^HLT_AK8PFJet[0-9]+.*BoostedDoubleB.*$
+    # - keep ^HLT_Diphoton.*NoPixelVeto.*$
+    - keep ^HLT_Diphoton.*
+    - keep ^HLT_Ele.*Ele.*$
+    - drop ^HLT_.*Tau3[mM]u.*$
+    # - drop ^HLT_Mu.*(Ele|Mu|Jpsi|Upsilon|NoFilters|IP).*
+    - drop ^HLT_Mu.*(Jpsi|Upsilon|NoFilters|IP).*
+
+skim_failed:
+  input_tree: Events
+  output_tree: EventsNotSelected
+  sel_ref: selection
+  invert_sel: true
+  column_filters:
+    - drop ^.*$
+    - keep ^LHE_.*$
+    - keep genWeight
+    - keep ^Pileup_(nPU|nTrueInt)$
+
+selection: "
+  auto ele_base_sel = Electron_pt > 7 && (abs(Electron_eta) < 2.5 || abs(Electron_eta+Electron_deltaEtaSC)<2.5) && abs(Electron_dz) < 1.0 && abs(Electron_dxy) < 0.5;
+  auto ele_mvaIso_sel = Electron_mvaIso_Fall17V2_WPL || Electron_mvaIso_WP90;
```

Comment:
Please run locally (as in readme step 4) to check that branch names are correct. In >= v10, branches were renamed, and, for example, `mvaIso` names do not have any more `Fall17V2` suffixes. There could be some other changes...
```shell
mkdir -p tmp && cd tmp
cmsEnv python3 $ANALYSIS_PATH/RunKit/nanoProdWrapper.py customise=NanoProd/NanoProd/customize.customize skimCfg=$ANALYSIS_PATH/NanoProd/config/skim_hzgamma.yaml maxEvents=100 sampleType=mc storeFailed=True era=Run2_2018 inputFiles=file:/eos/cms/store/group/phys_tau/kandroso/miniAOD_UL18/TTToSemiLeptonic.root writePSet=True skimSetup=skim skimSetupFailed=skim_failed createTar=False
cmsEnv $ANALYSIS_PATH/RunKit/crabJob.sh
```

Reply:
Thanks for the feedback!

- The minimal set of n-tuples we need actually does not include any data since we are currently planning on/in the process of deriving our corrections based on the central egamma tag-and-probe n-tuples. Thus, we only need the extra branches in MC to apply the corrections.
- As suggested, I've added skim requirements on photons, which should reduce the n-tuple size.
- The branch Electron_mvaIso_Fall17V2_WPL is available up through the version of NanoAODv12 that was used for the prompt 2023 NanoAODs. However, it is removed in the most recent version of NanoAODv12. The branches that do not contain "Fall17V2" are the Winter22 MVA, which unfortunately does not have a loose working point and I don't think has been studied with run 2. Trying to run locally as you suggest with the CMSSW generated by env.sh results in a SegFault because of this. I am not entirely familiar with how the customization is handled (via ex. customize.py), but I will try to investigate if there is a way to re-calcalculate the Fall17V2 MVA ID branch(es)

Reply:
After some investigation, I realized that in the central NanoAOD code[https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/electrons_cff.py] the mvaIso branches store different MVAs for run 2 and run 3. After switching the branch name to Electron_mvaIso_WPL, the local test runs without errors. Apologies for the confusion.

Reply:
Yes, no problem. It is always good to have another pair of eyes to cross-check. Thanks! It is also my understanding that the customisation for [Run 2](https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/electrons_cff.py#L365-L367) switches to `mvaIso` branches to `Fall17V2`.